### PR TITLE
limit image scale in pdf documents (fixes #2301)

### LINKED ISSUE #2301: scale insanely large images for PDF generation
By myself in #2288 

> I also noticed that (insanely) huge images do not display in PDF.
> Somewhat large images also render over the page-width.
> 
> This is not really an issue, since I would absolutely expect that it creates insanely huge images, but we can and should solve it nevertheless.
> 
> Or maybe this can get fixed in pdfmake's upstream?

### PR DESCRIPTION
**Pull request by tsiegleauq**
This solves the scale problem using a simple calculation.
Once there are proper ways to calculate the actual space (in pixel) on a page this can be enhanced.

@emanuelschuetze

### CODE REVIEW

File: openslides/core/static/js/core/site.js [hunk 1]
```diff
@@ -355,10 +355,27 @@ angular.module('OpenSlidesApp.core.site', [
                                     alreadyConverted.push(stack);
                                     break;
                                 case "img":
+                                    // pixel a4 in 72dpi: 595 x 842 pixel
+                                    // TODO: proper way to calculate the space left on the page.
+                                    // This requires further information
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/site.js]: Which information are still required?

File: openslides/core/static/js/core/site.js [hunk 2]
```diff
@@ -355,10 +355,27 @@ angular.module('OpenSlidesApp.core.site', [
...
+                                    var width = parseInt(element.getAttribute("width")),
+                                        height = parseInt(element.getAttribute("height"));
+
+                                    if (width > 590) {
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/site.js]: Please put the hard coded values into a variable, like so:
```javascript
var maxResolution = {
    width: 590,
    height: 842
}
if (width > maxResolution.width) {
    var scale = maxResolution.width / width
...
```

Of course you can rename `maxResolution` to something else.

File: openslides/core/static/js/core/site.js (hunk 1 above)
Comment (tsiegleauq) [inline openslides/core/static/js/core/site.js]: basically this is a todo comment to enhance it when more information about the document is present. All the information is required.
The comment also is meant to explain why we use 590 and 840.

File: openslides/core/static/js/core/site.js [hunk 3]
```diff
@@ -355,10 +355,32 @@ angular.module('OpenSlidesApp.core.site', [
...
                                 case "img":
+                                    // TODO: need a proper way to calculate the space
+                                    // left on the page.
+                                    // This requires further information
+                                    // A4 in 72dpi: 595px x 842px
+                                    var maxResolution = {
+                                        width: 590,
+                                        height: 840
+                                    },
+                                        width = parseInt(element.getAttribute("width")),
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/site.js]: Indention is broken.

File: openslides/core/static/js/core/site.js [hunk 4]
```diff
@@ -355,10 +355,32 @@ angular.module('OpenSlidesApp.core.site', [
...
+                                        height = parseInt(element.getAttribute("height"));
+
+                                    if (width > maxResolution.width) {
+                                        var scale = 590/width;
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/site.js]: use maxResolution.width instead of 590

File: openslides/core/static/js/core/site.js [hunk 5]
```diff
@@ -355,10 +355,32 @@ angular.module('OpenSlidesApp.core.site', [
...
+                                        width *= scale;
+                                        height *= scale;
+                                    }
+                                    if (height > maxResolution.height) {
+                                        var scale = 590/height;
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/site.js]: Here 590 is wrong.

File: openslides/core/static/js/core/site.js (hunk 3 above)
Comment (tsiegleauq) [inline openslides/core/static/js/core/site.js]: Does not seem broken to me.
were exactly?

Comment (tsiegleauq) [inline openslides/core/static/js/core/site.js]: you seem to be confused by the missing var. This conding style is consistent with the surounding coding stile in the same file.

```
var maxResolution = {
    width: 590,
    height: 840
},
    width = parseInt(element.getAttribute("width")),
    height = parseInt(element.getAttribute("height"));
```
is equivalent to:

```
var maxResolution = {
    width: 590,
    height: 840
};
var width = parseInt(element.getAttribute("width"));
var height = parseInt(element.getAttribute("height"));
```

Comment (normanjaeckel) [inline openslides/core/static/js/core/site.js]: Ok.

Comment (normanjaeckel) [inline openslides/core/static/js/core/site.js]: Maybe we should switch to CoffeeScript one day ...

**Comment by tsiegleauq** [comment]
@emanuelschuetze 
fixed the issue with the naming of the scale variable. Travis is happy now.

**Comment by emanuelschuetze** [comment]
Tested with huge photos. I reduced the width to render the image with correct right margin. And I added some css classes to limit the scaling in projector and detail view. Please merge my Pullrequest https://github.com/tsiegleauq/OpenSlides/pull/1. Then we can merge this one.

**Comment by emanuelschuetze** [comment]
Ok. Ready for review and merge.

**Comment by normanjaeckel** [comment]
Can you please rebase so that there are only two commits (remove the merge commit)? If not, I can merge this PR.

**Comment by emanuelschuetze** [comment]
@normanjaeckel Rebased. Ready for merge.

**Pull request merged**